shell/platform/android/test/io/flutter/plugin/platform/PlatformViewsControllerTest.java

```diff
@@ -385,6 +385,39 @@ public void createPlatformViewMessage__throwsIfViewHasParent() {
         });
   }
 
+  @Test
+  @Config(shadows = {ShadowFlutterJNI.class, ShadowPlatformTaskQueue.class})
+  public void setPlatformViewDirection__throwIfPlatformViewNotFound() {
+    PlatformViewsController platformViewsController = new PlatformViewsController();
+
+    int platformViewId = 0;
+    assertNull(platformViewsController.getPlatformViewById(platformViewId));
+
+    PlatformViewFactory viewFactory = mock(PlatformViewFactory.class);
+    PlatformView platformView = mock(PlatformView.class);
+
+    Context context = RuntimeEnvironment.application.getApplicationContext();
+    View androidView = new View(context);
+
+    when(platformView.getView()).thenReturn(androidView);
+    when(viewFactory.create(any(), eq(platformViewId), any())).thenReturn(platformView);
+    platformViewsController.getRegistry().registerViewFactory("testType", viewFactory);
+
+    FlutterJNI jni = new FlutterJNI();
+    attach(jni, platformViewsController);
+
+    // Simulate create call from the framework.
+    createPlatformView(jni, platformViewsController, platformViewId, "testType", /* hybrid=*/ true);
+    assertEquals(ShadowFlutterJNI.getResponses().size(), 1);
+
+    // Simulate set direction call from the framework.
```

Comment:
what about also adding verify(androidView, never()).setLayoutDirection(anyInt());  to ensure that it wasn't called prior to the next line?